service/authenticator_test.go

```diff
@@ -1,33 +1,710 @@
+	assert.NotNil(t, resp, validCase)
+	emailToken := resp.GetIdentification().GetToken()
+
+	cases := []struct {
+		desc       string
+		authHeader string
+		authPrefix string
+		input      string
+		isExpErr   bool
+		errStr     string
+	}{
+		{
+			"test missing auth header",
+			"",
+			consts.StrEmailTokenVerificationPrefix,
+			"",
+			true,
+			"rpc error: code = Unauthenticated desc = no \"authorization\" header in request",
+		},
+		{
+			"test wrong auth header",
+			"wrong",
+			consts.StrEmailTokenVerificationPrefix,
+			"",
+			true,
+			"rpc error: code = Unauthenticated desc = no \"authorization\" header in request",
+		},
+		{
+			"test missing auth prefix",
+			consts.StrMdBasicAuthHeader,
+			"",
+			"",
+			true,
+			"rpc error: code = Unauthenticated desc = missing \"Email Token \" prefix in \"authorization\" header",
+		},
+		{
+			"test wrong auth prefix",
+			consts.StrMdBasicAuthHeader,
+			"wrong",
+			"",
+			true,
+			"rpc error: code = Unauthenticated desc = missing \"Email Token \" prefix in \"authorization\" header",
+		},
+		{
+			"test fake email token",
+			consts.StrMdBasicAuthHeader,
+			consts.StrEmailTokenVerificationPrefix,
+			fakeAuthToken,
+			true,
+			"rpc error: code = Internal desc = no matching email token were found with given token",
+		},
+		{
+			validCase,
+			consts.StrMdBasicAuthHeader,
+			consts.StrEmailTokenVerificationPrefix,
+			emailToken,
+			false,
+			"",
+		},
+	}
+	for _, c := range cases {
+		header := metadata.New(map[string]string{
+			c.authHeader: c.authPrefix + c.input,
+		})
+		ctx := metadata.NewIncomingContext(context.Background(), header)
+		respCtx, err := tryEmailTokenVerification(ctx)
+		if c.isExpErr {
+			assert.EqualError(t, err, c.errStr, c.desc)
+		} else {
+			assert.Nil(t, err, c.desc)
+			assert.NotNil(t, respCtx, c.desc)
+		}
+	}
 
 }
 
 func TestTryTokenAuth(t *testing.T) {
+	nilIncomingCtxCase := "test for nil incoming context"
+	ctxNil, errNil := tryTokenAuth(nil)
+	assert.EqualError(t, errNil, consts.ErrNilContext.Error(), nilIncomingCtxCase)
+	assert.Equal(t, context.TODO(), ctxNil, nilIncomingCtxCase)
+
+	validCase := "test valid auth token"
+	validEmail := randomdata.Email()
+	validPassword := "Abcd!123@"
+	resp, err := userSvc.createUser(
+		&pblib.User{
+			FirstName:    randomdata.FirstName(randomdata.Male),
+			LastName:     randomdata.LastName(),
+			Email:        validEmail,
+			Password:     validPassword,
+			Organization: testOrg,
+		})
+	assert.Nil(t, err, validCase)
+	assert.NotNil(t, resp, validCase)
+	emailToken := resp.GetIdentification().GetToken()
+	errVerifyEmailToken := userSvc.verifyEmailToken(emailToken)
+	assert.Nil(t, errVerifyEmailToken, validCase)
+	validEnc := base64.StdEncoding.EncodeToString([]byte(validEmail + ":" + validPassword))
+	header := metadata.New(map[string]string{
+		consts.StrMdBasicAuthHeader: consts.StrBasicAuthPrefix + validEnc,
+	})
+	validIncomingCtx := metadata.NewIncomingContext(context.Background(), header)
```

Comment:
I think comments here to break this huge lines of codes is helpful b/c i'm having hard time reading it.